### DISCUSSION TITLE
Not able to login via auth

### DISCUSSION THREAD

**Discussion by Shub-Gautam** [umputun/remark42#1457] [Q&A]
Receive {"error":"failed to get token"} after signing in through google auth

Docker-compose.yml

version: "2"

    services:
      remark:
        build: .
        image: umputun/remark42:latest
        container_name: "remark42"
        hostname: "remark42"
        restart: always

        logging:
          driver: json-file
          options:
            max-size: "10m"
            max-file: "5"
    
        ports:
          - "8081:8081"
          - "8080:8080"
          - "8000:8000"
    
        environment:
          - REMARK_URL=http://localhost:8080
          - SECRET=12345
          - STORE_BOLT_PATH=/srv/var/db
          - BACKUP_PATH=/srv/var/backup
          - AUTH_GOOGLE_CID=<ID>
          - AUTH_GOOGLE_CSEC=<Key>
          - DEBUG=true
          - AUTH_ANON=true
          - AUTH_SAME_SITE=none
    
        volumes:
          - ./var:/srv/var

![New Issue · umputun_remark42 and 138 more pages - Personal - Microsoft​ Edge 31-07-2022 20_24_04](https://user-images.githubusercontent.com/72097846/182032348-b946115e-480a-4b8d-8089-1676eea734ea.png)

**Comment by paskal** [comment]
Could you please try the master Docker image tag and tell us if the issue persists? I think most likely we fixed it since then but haven't released a bugfix version yet.

**Comment by Shub-Gautam** [comment]
Its working, thanks !

**Comment by Shub-Gautam** [comment]
It is not working on some different domain even after setting `AUTH_SAME_SITE=none` in docker-compose.yml
Is there any new changes in this flow in master image on docker ?

**Comment by paskal** [comment]
Please see the following doc. It should have answers to your questions: https://remark42.com/docs/manuals/separate-domain/

**Comment by Shub-Gautam** [comment]
Tried it but the issue is still there, interestingly when I set  `AUTH_SAME_SITE=none` , auth on which remark server is hosted (http://localhost:8080 in my case) also shows the error page when you try to log in via auth (Google auth) on /web page (http://localhost:8080/web), but when you refresh the page (http://localhost:8080/web), it shows your logged in profile,

================ Observations ==============
 - `AUTH_SAME_SITE=none`  => Showing an error page after you login via auth on page /web (REMARK_URL/web), but after page refresh your profile is visible on the page, Auth not working on any other domain
 - Default Configuration => Auth working normally on the /web page (REMARK_URL/web), but not working on any other domain
 
docker-compose.yml

        version: "2"
        
        services:
          remark:
            build: .
            image: umputun/remark42:master
            container_name: "remark42"
            hostname: "remark42"
            restart: always
        
            logging:
              driver: json-file
              options:
                max-size: "10m"
                max-file: "5"
        
            ports:
              - "8080:8080"
        
            environment:
              - REMARK_URL=http://localhost:8080
              - SECRET=12345
              - SITE=remark
              - STORE_BOLT_PATH=/srv/var/db
              - BACKUP_PATH=/srv/var/backup
              - AUTH_GOOGLE_CID=<Key>
              - AUTH_GOOGLE_CSEC=<key>
              - AUTH_SAME_SITE=none
        
            volumes:
              - ./var:/srv/var

**Comment by paskal** [comment]
Have you set the `ALLOWED_HOSTS` as the documentation suggests? Have you read that the auth wouldn't work if the domain name is different (like remark42.example1.org and blog.example2.com)? Are you running Remark42 on the same domain as the site on which you would be using it?

**Comment by Shub-Gautam** [comment]
Yes I even tried doing this, but the issue is still there
          
         version: "2"
          
          services:
            remark:
              build: .
              image: umputun/remark42:master
              container_name: "remark42"
              hostname: "remark42"
              restart: always
          
              logging:
                driver: json-file
                options:
                  max-size: "10m"
                  max-file: "5"
          
              ports:
                - "8080:8080"
          
              environment:
                - REMARK_URL=http://localhost:8080
                - SECRET=12345
                - SITE=remark
                - STORE_BOLT_PATH=/srv/var/db
                - BACKUP_PATH=/srv/var/backup
                - AUTH_GOOGLE_CID=<Key>
                - AUTH_GOOGLE_CSEC=<Key>
                - ALLOWED_HOSTS='self',http://localhost:5500,http://localhost:8000
                - AUTH_SAME_SITE=none
          
              volumes:
                - ./var:/srv/var

**Comment by paskal** [comment]
Can you please tell me once more what you are trying to achieve, what the expected result and what you see in reality?

**Comment by Shub-Gautam** [comment]
I want to add comment functionality on my site which is running on localhost:8000 and my remark42 server running on localhost:8080 configured by docker-compose file as shown below

     version: "2"
      
      services:
        remark:
          build: .
          image: umputun/remark42:master
          container_name: "remark42"
          hostname: "remark42"
          restart: always
      
          logging:
            driver: json-file
            options:
              max-size: "10m"
              max-file: "5"
      
          ports:
            - "8080:8080"
      
          environment:
            - REMARK_URL=http://localhost:8080
            - SECRET=12345
            - SITE=remark
            - STORE_BOLT_PATH=/srv/var/db
            - BACKUP_PATH=/srv/var/backup
            - AUTH_GOOGLE_CID=<Key>
            - AUTH_GOOGLE_CSEC=<Key>
            - ALLOWED_HOSTS='self',http://localhost:5500,http://localhost:8000
            - AUTH_SAME_SITE=none
      
          volumes:
            - ./var:/srv/var


===== Observation =====

I want my users to login through google auth, but when I click on "Sign In with Google" after loging in a page appears with this error  

![Screenshot 2022-08-01 225957](https://user-images.githubusercontent.com/72097846/182208008-ba4f2e78-b777-4fa0-ac2f-86c9a243d242.jpg)


- Anonymous login also does not work on different port when you setup remark42 by above docker-compose file
- If you setup remark42 by default parameters (Not passing AUTH_SAME_SITE,ALLOWED_HOSTS in environment) auth and anonymous login working just fine only on port on which remark42 is running (in my case localhost:8080) but not on any other port (in my case localhost:8000)
- If you setup remark42 by passing these parameters
`ALLOWED_HOSTS='self',http://localhost:5500,http://localhost:8000
 AUTH_SAME_SITE=none`
 it should allow google auth on ports 8080(self), 5500 and 8000 but still getting the same error page.

**Reply by paskal** [reply]
I'll test it, but first thing first: you should use 127.0.0.1 instead of localhost as frontend is known to work only with that address and not localhost, as well as the dev auth provider on the backend side.

**Reply by paskal** [reply]
Anonymous auth works for me, with port 8000 exposed as `- "8000:8080"` (e.g. directly from the container, no web server in between). Google doesn't work as a redirect to localhost gives me an error on the Google side.

I think what is missing for you is `docker pull umputun/remark42:master` as you're likely using an outdated image. Latest image produces the following line in the log on start: ` remark42 master-27b28db-20220801T10:55:59`.

**Comment by umputun** [comment]
as this is not an actual bug report, but rather configuration issue, I'll move it to discussions
